### PR TITLE
SetterMustExistRule - Ignore Synthetic Fields

### PR DESCRIPTION
**Pull request by scampbell5**
- Added a check in the SetterMustExistRule to ignore synthetic fields

### CODE REVIEW

File: src/main/java/com/openpojo/validation/rule/impl/SetterMustExistRule.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
   public void evaluate(final PojoClass pojoClass) {
     for (PojoField fieldEntry : pojoClass.getPojoFields()) {
-      if (!fieldEntry.isFinal() && !fieldEntry.hasSetter()) {
+      if (!fieldEntry.isFinal() && !fieldEntry.hasSetter() && fieldEntry.isSynthetic()) {
```
Comment (mustaphazorgati) [inline src/main/java/com/openpojo/validation/rule/impl/SetterMustExistRule.java:37]:
```suggestion
      if (!fieldEntry.isSynthetic() && !fieldEntry.isFinal() && !fieldEntry.hasSetter()) {
```

Comment (mustaphazorgati) [reply src/main/java/com/openpojo/validation/rule/impl/SetterMustExistRule.java:37]: you forgot to negate the `isSynthetic`.

Comment (scampbell5) [reply src/main/java/com/openpojo/validation/rule/impl/SetterMustExistRule.java]: Thank you :)

**Comment by mustaphazorgati** [comment]
@oshoukry can you please have a look at this? This seems to be a little addition which is pretty cool. 
As of right now we had to duplicate that class in order to exclude synthetic fields (because we use AspectJ)

**Comment by oshoukry** [comment]
rejected this code change as it doesn't exercise the failure due to the fact that the synthetic generated field is already skipped with !final condition.  Issue fixed on another checking, see: https://github.com/OpenPojo/openpojo/commit/23678f8859bdc767c20861dd143a0af1f3f2fc9f

**Pull request closed without merging**